src/privateSubscription.js

```diff
@@ -1,106 +1,256 @@
 const EventEmitter = require("events");
 
-function PrivateSubscription(web3, privacyGroupId, filter) {
-  this.privacyGroupId = privacyGroupId;
-  this.filter = filter;
+const Protocol = {
+  HTTP: "HTTP",
+  WEBSOCKET: "WebSocket"
+};
 
-  this.web3 = web3;
-  this.filterId = null;
+const Event = {
+  CONNECTED: "connected",
+  DATA: "data",
+  ERROR: "error"
+};
 
-  this.timer = undefined;
-  this.getPast = false;
+/**
+ * Manage a specific type of subscription
+ * @param {PrivateSubscription} subscription
+ */
+function SubscriptionManager(subscription) {
+  this.subscription = subscription;
+  this.web3 = subscription.web3;
 
   return this;
 }
 
-// get functions from EventEmitter
-PrivateSubscription.prototype = Object.create(EventEmitter.prototype);
+/**
+ * Manage creating/destroying filter and polling for new logs
+ * using `priv_getFilterChanges`
+ * @param {PrivateSubscription} subscription
+ */
+function PollingSubscription(subscription, pollingInterval) {
+  SubscriptionManager.call(this, subscription);
 
-PrivateSubscription.prototype.subscribe = async function subscribe() {
+  this.privacyGroupId = subscription.privacyGroupId;
+  this.filter = subscription.filter;
+  this.timeout = null;
+  // How frequently to poll for new logs, in milliseconds
+  this.pollingInterval = pollingInterval || 1000;
+
+  return this;
+}
+PollingSubscription.prototype = Object.create(SubscriptionManager.prototype);
+PollingSubscription.prototype.constructor = PollingSubscription;
+
+PollingSubscription.prototype.subscribe = async function subscribe(
```

Comment:
When we create a filter subscription, are we brining the past matching log messages (basically, starting the filter with a `priv_getFilterLogs` and then subsequent `priv_getFilterChanges`?).

Reply:
I'm actually using `priv_getLogs` in both cases to get the past logs, because it seemed less complex to use the same method. Is there a reason to prefer `priv_getFilterLogs`?

Reply:
We need to ensure that we will never return logs that don't match the filter. Using `priv_getFilterLogs` ensures we are applying the filter query.

Because the filters are maintained in the server-side, one could create a filter in the server, and had it modified. In this case, using getLogs with the "original" filter could return unexpected results.

I don't think it is a big deal, but I think using the filter will help us know that it will always respect the query stored on the server. And it shouldn't be a big change, am I right?